ApiDoctor.Validation/OData/ODataParser.cs

```diff
@@ -187,49 +189,66 @@ public static async Task<EntityFramework> ParseEntityFrameworkFromFileAsync(stri
+            resources.AddRange(from et in schema.EntityTypes select ResourceDefinitionFromType(schema, otherSchema, et, issues, metadataValidationConfigs));
 
             return resources;
         }
 
-        private static ResourceDefinition ResourceDefinitionFromType(Schema schema, IEnumerable<Schema> otherSchema, ComplexType ct, IssueLogger issues)
+        private static ResourceDefinition ResourceDefinitionFromType(Schema schema, IEnumerable<Schema> otherSchema, ComplexType ct, IssueLogger issues, MetadataValidationConfigs metadataValidationConfigs)
         {
-            var annotation = new CodeBlockAnnotation() { ResourceType = string.Concat(schema.Namespace, ".", ct.Name), BlockType = CodeBlockType.Resource };
-            var json = BuildJsonExample(ct, otherSchema);
+            var metadataNamespace = metadataValidationConfigs?.ModelConfigs?.AliasNamespace?? schema.Namespace;
+            var resourceType = string.Concat(metadataNamespace + "." + ct.Name);
+
+            if (!metadataValidationConfigs.ModelConfigs.ValidateNamespace)
+            {
+                resourceType = ct.Name;
+            }
+
+            var annotation = new CodeBlockAnnotation() { ResourceType = resourceType, BlockType = CodeBlockType.Resource };
+            var json = BuildJsonExample(ct, otherSchema, metadataValidationConfigs);
             ResourceDefinition rd = new JsonResourceDefinition(annotation, json, null, issues);
             return rd;
         }
 
-        public static string BuildJsonExample(ComplexType ct, IEnumerable<Schema> otherSchema)
+        public static string BuildJsonExample(ComplexType ct, IEnumerable<Schema> otherSchema, MetadataValidationConfigs metadataValidationConfigs = null)
         {
-            Dictionary<string, object> dict = BuildDictionaryExample(ct, otherSchema);
+            Dictionary<string, object> dict = BuildDictionaryExample(ct, otherSchema, metadataValidationConfigs);
             return JsonConvert.SerializeObject(dict, Newtonsoft.Json.Formatting.Indented);
         }
 
-        private static Dictionary<string, object> BuildDictionaryExample(ComplexType ct, IEnumerable<Schema> otherSchema)
+        private static Dictionary<string, object> BuildDictionaryExample(ComplexType ct, IEnumerable<Schema> otherSchema, MetadataValidationConfigs metadataValidationConfigs = null)
         {
             Dictionary<string, object> propertyExamples = new Dictionary<string, object>();
 
             if (!string.IsNullOrWhiteSpace(ct.Namespace))
             {
-                propertyExamples.Add("@odata.type", $"{ct.Namespace}.{ct.TypeIdentifier}");
+                var metadataNamespace = metadataValidationConfigs?.ModelConfigs?.AliasNamespace ?? ct.Namespace;
+                var resourceTypeIdentifier = !string.IsNullOrEmpty(metadataValidationConfigs?.ModelConfigs?.AliasNamespace)
+                    ? $"{metadataNamespace}.{ct.TypeIdentifier}"
+                    : ct.TypeIdentifier;
```

Comment:
Missing concatenation with `ct.Namespace`?  (shouldn't it be `ct.Namespace + "." + ct.TypeIdentifier` like in `ResourceDefinitionFromType`?)

Reply:
If that's the case also make this code more consistent with `ResourceDefinitionFromType` - actually you could have a helper function that generates a type identifier from metadataValidationConfigs + default namespace + type name.